### PR TITLE
[feat](iceberg)Supports using `rest` type catalog to read tables in unity catalog

### PR DESCRIPTION
**Pull request by wuwenchi**
### What problem does this PR solve?

1. We now support using the `rest` type catalog to read tables in the unity catalog (https://github.com/unitycatalog/unitycatalog).
2. When reading the parquet file on the be side, we find the corresponding column name based on the column id, which naturally supports the column rename function.

example：
```
CREATE CATALOG `uc3`
PROPERTIES (
"type"  =  "iceberg",
"iceberg.catalog.type"  =  "rest",
"uri"  =  "http://127.0.0.1:8080/api/2.1/unity-catalog/iceberg",
"external_catalog.name" = "unity"  --- catalog name in unity catalog
);
```
### Release Note

[feat](catalog) support iceberg unity catalog

### Check List (For Author)

- Test 
    - [x] Regression test
    - [ ] Unit Test
    - [ ] Manual test (add detailed scripts or steps below)
    - [ ] No need to test or manual test. Explain why:
        - [ ] This is a refactor/code format and no logic has been changed.
        - [ ] Previous test can cover this change.
        - [ ] No code files have been changed.
        - [ ] Other reason 

- Behavior changed:
    - [x] No.
    - [ ] Yes. 

- Does this need documentation?
    - [ ] No.
    - [x] Yes. 

### Check List (For Reviewer who merge this PR)

- [x] Confirm the release note
- [x] Confirm test cases
- [ ] Confirm document
- [x] Add branch pick label

### CODE REVIEW

**Comment by wuwenchi** [comment]
run buildall

**Comment by wuwenchi** [comment]
run buildall

File: be/src/vec/exec/format/table/iceberg_reader.h [hunk 1]
```diff
@@ -218,7 +218,7 @@ class IcebergParquetReader final : public IcebergTableReader {
         parquet_reader->set_delete_rows(&_iceberg_delete_rows);
     }
 
-    Status _gen_col_name_maps(std::vector<tparquet::KeyValue> parquet_meta_kv);
+    Status _gen_col_name_maps(FieldDescriptor field_desc);
```
[the remark this answers is not in the document]
Comment (wuwenchi) [reply be/src/vec/exec/format/table/iceberg_reader.h:221]: The orc format is originally associated according to the id, so there is no need to modify it.

**Comment by wuwenchi** [comment]
run buildall

**Comment by wuwenchi** [comment]
run buildall

**Comment by wuwenchi** [comment]
run buildall

**Comment by wuwenchi** [comment]
run buildall

**Comment by wuwenchi** [comment]
run buildall

**Review by morningman — DISMISSED** [review]
LGTM

**Review by hubgeter — APPROVED** [review]

**Review by suxiaogang223 — APPROVED** [review]

File: be/src/io/file_factory.cpp [hunk 2]
```diff
@@ -147,8 +147,12 @@ Result<io::FileReaderSPtr> FileFactory::create_file_reader(
...
     case TFileType::FILE_LOCAL: {
         io::FileReaderSPtr file_reader;
-        RETURN_IF_ERROR_RESULT(io::global_local_filesystem()->open_file(
-                file_description.path, &file_reader, &reader_options));
+        std::string final_file = file_description.path;
+        if (final_file.rfind("file:", 0) == 0) { // Check if 'str' starts with 'prefix'
+            final_file = final_file.substr(5);
+        }
+        RETURN_IF_ERROR_RESULT(io::global_local_filesystem()->open_file(final_file, &file_reader,
+                                                                        &reader_options));
```
Comment (dataroaring) [inline be/src/io/file_factory.cpp:155]: file:: prefix is duplicated with type? Maybe we should parse when get type and check if they are same.

File: be/src/io/fs/local_file_system.cpp [hunk 3]
```diff
@@ -471,4 +471,35 @@ Status LocalFileSystem::permission_impl(const Path& file, std::filesystem::perms
     return Status::OK();
 }
 
+Path LocalFileSystem::convertToAbsPath(const Path& inputPathString) {
```
Comment (gavinchou) [inline be/src/io/fs/local_file_system.cpp:474]: `convertToAbsPath` code style should be `lower_under_score`

File: be/test/io/fs/local_file_system_test.cpp [hunk 4]
```diff
@@ -417,4 +417,11 @@ TEST_F(LocalFileSystemTest, TestGlob) {
     EXPECT_TRUE(io::global_local_filesystem()->delete_directory(path).ok());
 }
 
+TEST_F(LocalFileSystemTest, TestConvertToAbsPath) {
```
Comment (gavinchou) [inline be/test/io/fs/local_file_system_test.cpp:420]: add more negative cases e.g.
"hdfds://fdkslaf"
"abc/d/c/" 
"s3://"
"file"
"feil:"

File: be/src/io/fs/local_file_system.cpp [hunk 5]
```diff
@@ -471,4 +471,35 @@ Status LocalFileSystem::permission_impl(const Path& file, std::filesystem::perms
...
+    std::string pathString = inputPathString;
+
+    // Initialize scheme and authority
+    std::string scheme;
+    std::string authority;
+
+    size_t start = 0;
+
+    // Parse URI scheme, if any
+    size_t colon = pathString.find(':');
```
Comment (gavinchou) [inline be/src/io/fs/local_file_system.cpp:484]: why not use `std::string::starts_with()` ?

File: be/src/io/fs/local_file_system.h [hunk 6]
```diff
@@ -34,6 +34,8 @@ class LocalFileSystem final : public FileSystem {
 public:
     ~LocalFileSystem() override;
 
+    static Path convert_to_abs_path(const Path& path);
```
Comment (gavinchou) [inline be/src/io/fs/local_file_system.h:37]: add comment for this public static function. how does it behave
we have to emphasize that what is the path and what is the abs path.
what if the path is invalid?

**Comment by wuwenchi** [comment]
run buildall

**Comment by wuwenchi** [comment]
run buildall

**Comment by wuwenchi** [comment]
run buildall

**Comment by wuwenchi** [comment]
run buildall

**Comment by wuwenchi** [comment]
run buildall

**Comment by wuwenchi** [comment]
run buildall

**Review by morningman — DISMISSED** [review]
LGTM

File: be/src/io/fs/local_file_system.cpp [hunk 7]
```diff
@@ -471,4 +471,46 @@ Status LocalFileSystem::permission_impl(const Path& file, std::filesystem::perms
...
+        scheme = path_str.substr(0, colon);
+        if (scheme != "file") {
+            return Status::InternalError(
+                    "Only supports `file` type scheme, like 'file:///path', 'file:/path'.");
+        }
+        start = colon + 1;
+    }
+
+    // Parse URI authority, if any
+    if (path_str.compare(start, 2, "//") == 0 && path_str.length() - start > 2) {
```
Comment (gavinchou) [inline be/src/io/fs/local_file_system.cpp:504]: add example here

**Comment by wuwenchi** [comment]
run buildall

**Comment by wuwenchi** [comment]
run feut

**Review by gavinchou — APPROVED** [review]

**Review by morningman — APPROVED** [review]

**Pull request merged**